docs/pin.rst

```diff
@@ -0,0 +1,112 @@
+Input/Output Pins
+*****************
+
+.. py:module:: microbit
+
+The pins are your board's way to communicate with external devices connected to
+it. There are 19 pins for your disposal, numbered 0-16 and 19-20. Pins 17 and
+18 are not available.
+
+.. image:: pinout.png
+
+Those pins are available as attributes on the ``microbit``
+module:``microbit.pin0`` - ``microbit.pin20``.
+
++-----+---------+----------+
+| Pin | Type    | Function |
++=====+=========+==========+
+|  0  | Touch   | Pad 0    |
++-----+---------+----------+
+|  1  | Touch   | Pad 1    |
++-----+---------+----------+
+|  2  | Touch   | Pad 2    |
++-----+---------+----------+
+|  3  | Analog  | Column 1 |
++-----+---------+----------+
+|  4  | Digital | Button A |
++-----+---------+----------+
+|  5  | Analog  | Column 2 |
++-----+---------+----------+
+|  6  | Digital | Row 2    |
++-----+---------+----------+
+|  7  | Digital | Row 1    |
++-----+---------+----------+
+|  8  | Digital |          |
++-----+---------+----------+
+|  9  | Digital | Row 3    |
++-----+---------+----------+
+|  10 | Analog  | Column 3 |
++-----+---------+----------+
+|  11 | Digital | Button B |
++-----+---------+----------+
+|  12 | Digital |          |
++-----+---------+----------+
+|  13 | Digital | SPI MOSI |
++-----+---------+----------+
+|  14 | Digital | SPI MISO |
++-----+---------+----------+
+|  15 | Digital | SPI SCK  |
++-----+---------+----------+
+|  16 | Digital |          |
++-----+---------+----------+
++-----+---------+----------+
+|  19 | Digital | I2C SCL  |
++-----+---------+----------+
+|  20 | Digital | I2C SDA  |
++-----+---------+----------+
+
+The above table summarizes the pins available, their types (see below) and what
+they are internally connected to.
```

Comment:
We will probably need some explanations about the functions of those pins, with examples etc.